File: management-ui/backend/app/schemas/service.py
from pydantic import BaseModel, Field
from typing import List, Dict, Optional, Any
from enum import Enum

class ServiceStatus(str, Enum):
    RUNNING = "running"
    STOPPED = "stopped"
    STARTING = "starting"
    STOPPING = "stopping"
    RESTARTING = "restarting"
    ERROR = "error"
    NOT_CREATED = "not_created"

class HealthStatus(str, Enum):
    HEALTHY = "healthy"
    UNHEALTHY = "unhealthy"
    STARTING = "starting"
    NONE = "none"
    UNKNOWN = "unknown"

class ServiceInfo(BaseModel):
    name: str
    status: ServiceStatus
    health: HealthStatus
    image: Optional[str] = None
    container_id: Optional[str] = None
    ports: List[str] = []
    group: str
    compose_file: str
    profiles: List[str] = []
    has_healthcheck: bool = False
    depends_on: Dict[str, str] = {}

class ServiceListResponse(BaseModel):
    services: List[ServiceInfo]
    total: int

class ServiceDetailResponse(ServiceInfo):
    environment: Dict[str, str] = {}
    created: Optional[str] = None
    dependents: List[str] = []
    dependencies: List[str] = []

class ServiceActionRequest(BaseModel):
    profile: Optional[str] = None
    environment: str = "private"
    force: bool = False  # Force stop even if has dependents

class ServiceActionResponse(BaseModel):
    success: bool
    message: str
    service: str
    action: str
    output: Optional[str] = None
    error: Optional[str] = None

class DependencyWarning(BaseModel):
    service: str
    dependents: List[str]
    message: str

class DependencyGraphResponse(BaseModel):
    nodes: List[Dict[str, Any]]
    edges: List[Dict[str, Any]]

class ServiceGroupInfo(BaseModel):
    id: str
    name: str
    description: str
    services: List[str]
    running_count: int = 0
    total_count: int = 0

class ServiceGroupsResponse(BaseModel):
    groups: List[ServiceGroupInfo]

File: management-ui/frontend/src/types/service.ts
export type ServiceStatus = 'running' | 'stopped' | 'starting' | 'stopping' | 'restarting' | 'error' | 'not_created';
export type HealthStatus = 'healthy' | 'unhealthy' | 'starting' | 'none' | 'unknown';

export interface ServiceInfo {
  name: string;
  status: ServiceStatus;
  health: HealthStatus;
  image: string | null;
  container_id: string | null;
  ports: string[];
  group: string;
  compose_file: string;
  profiles: string[];
  has_healthcheck: boolean;
  depends_on: Record<string, string>;
}

export interface ServiceDetail extends ServiceInfo {
  environment: Record<string, string>;
  created: string | null;
  dependents: string[];
  dependencies: string[];
}

export interface ServiceGroup {
  id: string;
  name: string;
  description: string;
  services: string[];
  running_count: number;
  total_count: number;
}

export interface ServiceActionRequest {
  profile?: string;
  environment?: string;
  force?: boolean;
}

export interface ServiceActionResponse {
  success: boolean;
  message: string;
  service: string;
  action: string;
  output?: string;
  error?: string;
}
